backend/ftp/location.go

```diff
@@ -156,8 +157,8 @@ func (l *Location) Exists() (bool, error) {
 		return false, err
 	}
 
-	for _, entry := range entries {
-		if entry.Name == path.Base(l.Path()) && entries[0].Type == _ftp.EntryTypeFolder {
+	for i := range entries {
+		if entries[i].Name == locBasename && entries[i].Type == _ftp.EntryTypeFolder {
```

Comment:
entries[0] was the culprit here... only checking the first entry's type, resulting Exists returning false, then MoveToFile tried to create the dir (which already existed).